pkgs/top-level/lua-packages.nix

```diff
@@ -215,6 +215,30 @@ let
     '';
   };
 
+  luaiconv = buildLuaPackage rec {
+    name = "luaiconv-${version}";
+    version = "7";
+    src = fetchFromGitHub {
+      owner = "ittner";
+      repo = "lua-iconv";
+      rev = "e8d34024a6b185a759733915f116cc5588550261";
+      sha256 = "0rd76966qlxfp8ypkyrbif76nxnm1acclqwfs45wz3972jsk654i";
+    };
+
+    buildInputs = [];
+    meta = {
+      platforms = stdenv.lib.platforms.unix;
+      license = stdenv.lib.licenses.mit;
+      description = "Lua bindings for POSIX iconv";
+      maintainers = [ maintainers.richardipsum ];
+    };
+    preBuild = ''
+      makeFlagsArray=(
+        INSTALL_PATH="$out/lib/lua/${lua.luaversion}"
+      );
+    '';
```

Comment:
Use `makeFlags`, you don't need `preBuild` for this.

Reply:
I'm not familiar with that, but I'll look into it, thanks. :cat:

Reply:
Everything else in this file seems to use `preBuild` :confused:

Reply:
Yeah, buildLuaPackage does not accommodate `makeFlags`… You may use this instead:
`installFlags = [ "INSTALL_PATH=$(out)/lib/lua/${lua.luaversion}" ];`

Reply:
@richardipsum `makeFlags = [ INSTALL_PATH="$out/lib/lua/${lua.luaversion}" ];`

Reply:
Looking at my build log I don't think this is required, currently, I see:

    make[1]: Leaving directory '/tmp/nix-build-lua5.1-luaiconv-7.drv-0/lua-iconv-e8d34024a6b185a759733915f116cc5588550261-src'
    install -D -s iconv.so //nix/store/1khy7gdmz7fkim58ff05h7j2aflxb7hg-lua5.1-luaiconv-7/lib/lua/5.1/iconv.so
    post-installation fixup
    shrinking RPATHs of ELF executables and libraries in /nix/store/1khy7gdmz7fkim58ff05h7j2aflxb7hg-lua5.1-luaiconv-7
    shrinking /nix/store/1khy7gdmz7fkim58ff05h7j2aflxb7hg-lua5.1-luaiconv-7/lib/lua/5.1/iconv.so

and when I modify the contents of `INSTALL_PATH` by appending a test string `/TESTSTRING`, then I see:

    make[1]: Leaving directory '/tmp/nix-build-lua5.1-luaiconv-7.drv-0/lua-iconv-e8d34024a6b185a759733915f116cc5588550261-src'
    install -D -s iconv.so //nix/store/m9py1qwly24gf1mk5k7wqm3vdkq4dzab-lua5.1-luaiconv-7/lib/lua/5.1/TESTSTRING/iconv.so
    post-installation fixup
    shrinking RPATHs of ELF executables and libraries in /nix/store/m9py1qwly24gf1mk5k7wqm3vdkq4dzab-lua5.1-luaiconv-7
    shrinking /nix/store/m9py1qwly24gf1mk5k7wqm3vdkq4dzab-lua5.1-luaiconv-7/lib/lua/5.1/TESTSTRING/iconv.so

This gives me some confidence that `INSTALL_PATH` is being picked up, other lua packages also set their install dir this way, as I say I really have just adopted the existing conventions used in this file.

Reply:
@vyp This is a little deficiency of `buildLuaPackage`, it overrides `makeFlags` with its `makeFlagsArray` instead of extending them. We could fix it…

Reply:
@richardipsum `makeFlagsArray` in `preBuild` is certainly correct, it's just not common outside of lua-packages when `makeFlags` is sufficient. Currently lua-packages have to resort to this (or, rather, set `installFlags` instead of `makeFlags`) because `buildLuaPackage` itself defines `makeFlagsArray` in its `preBuild`, but it does not seem to have a reason not to use `makeFlags` instead.

Reply:
@orivej I see, noted, it's unclear to me what that means for the patches I have in flight. I would hazard a guess that were this to be fixed it should be done for all lua-packages to maintain consistency?